source/store/CaseContext.js

```diff
@@ -1,139 +1,67 @@
-        })
-        .then(response => callback(response));
+    async function loadCases(callback = () => {}) {
+      dispatch(await fetch(user, callback));
     },
-    [user]
+    [dispatch, user]
   );
 
   useEffect(() => {
     if (user) {
-      fetchCases(response => {
-        setCurrentCase(findLatestCase(response));
-      });
+      fetchCases();
     }
-  }, [fetchCases, user]);
-
-  const getCase = caseId => cases.find(c => c.id === caseId);
-
-  /**
-   * Function for sending a post request towards the case api endpoint.
-   * Can be used with  callback if something is to be done with the response object.
-   * @param {obj} data a object consiting of case user inputs.
-   * @param {string} formId the id of the form corresponding to the new case
-   * @param {func} callback what should happen after the new case have been created.
-   */
-  const createCase = async (data, formId, callback = response => {}) => {
-    const body = {
-      personalNumber: parseInt(user.personalNumber),
-      status: 'completed',
-      type: 'VIVA_CASE',
-      data: data || {},
-      formId,
-    };
-    // TODO: Remove Auhtorization header when token authentication works as expected.
-    post('/cases', JSON.stringify(body), {
-      Authorization: parseInt(user.personalNumber),
-    })
-      .then(response => {
-        setCurrentCase(response.data.data);
-        setFetching(false);
-        return response.data.data;
-      })
-      .then(newCase => callback(newCase));
-  };
-
-  /**
-   * Function for sending a put request towards the case api endpoint, updating the currently active case
-   * @param {obj} data a object consiting of case user inputs.
-   */
-  const updateCurrentCase = async (data, status, currentStep) => {
-    const body = {
-      status,
-      data,
-      currentStep,
-    };
-    // TODO: Remove Auhtorization header when token authentication works as expected.
-
-    try {
-      await put(`/cases/${currentCase.id}`, JSON.stringify(body), {
-        Authorization: parseInt(user.personalNumber),
-      }).then(res => {
-        const { caseId: id, ...other } = res.data.data;
-        const updatedCase = { id, ...other, updatedAt: Date.now(), ...currentCase };
-        setCurrentCase(updatedCase);
-
-        const newCases = JSON.parse(JSON.stringify(cases));
-        newCases[cases.findIndex(c => c.id === currentCase.id)] = updatedCase;
-        setCases(newCases);
-      });
-    } catch (error) {
-      console.log(`Update current case error: ${error}`);
-    }
-  };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [user]);
 
   return (
-    <CaseContext.Provider
-      value={{
-        cases,
-        currentCase,
-        getCase,
-        setCurrentCase,
-        createCase,
-        updateCurrentCase,
-        fetchCases,
-        fetching,
-      }}
-    >
-      {children}
-    </CaseContext.Provider>
+    <CaseState.Provider value={{ cases: state?.cases || {}, getCase }}>
```

Comment:
Do you need to do the check (state?.cases || {}) here? Should be enough to set the initialState of the reducer to an empty object and just pass state.cases

Reply:
Fixed